getgauge/impl_loader.py

```diff
@@ -84,3 +88,6 @@ def _get_version():
     json_data = open(PLUGIN_JSON).read()
     data = json.loads(json_data)
     return data[VERSION]
+
+def _is_step_impl_class(file):
+    return len([ step for step in PythonFile.parse(file).iter_steps() ]) > 0
```

Comment:
Checking only for `step` might not be enough. What about hooks defined in a class file and accessing the `self` parameter. The execution might not break cause we pass one parameter to hooks (ExecutionContext), but the parameter passed will not be correct (will be context instead of `self`).

Same goes for other APIs exposed from the getgauge package. We have to check for all of them.

The docs need to be updated with warning/note saying any class using gauge replated APIs should not have a constructor with a parameter.

/cc @sriv @nehashri